### DISCUSSION TITLE
alternative way to specify image policies

### DISCUSSION THREAD

**Discussion by masterkain** [fluxcd/flux2#2830] [Q&A] [answered]
https://fluxcd.io/docs/guides/image-update/#configure-image-update-for-custom-resources

I'm generating some HelmRelease programmatically via Ruby YAML.
I'm finding impossible to add a comment on a line and save the file. I might resort to using an end line value such as `! {"$imagepolicy": "flux-system:podinfo:name"}` and perform a substitution via sed or similar, but it's another step in an already complicated pipeline.

I guess there are no other alternative way to obtain the same result (in flux 1 this was different), but I'd like to ask if that's the only way and/or is there a plan to specify policies in an alternative way.

**Comment by pjbgf** [comment, marked as answer]
I am not aware of any other way that the image automation controller can identify what policy to pick up apart from the comment in the YAML file.

Maybe other folks know some Ruby library that may allow for you to add comments programmatically. (@kingdonb)

**Comment by 4F2E4A2E** [comment]
ytt and kustomize are cutting off comments by default.
